api/s2n.h

```diff
@@ -66,6 +66,15 @@ typedef enum {
     S2N_EXTENSION_CERTIFICATE_TRANSPARENCY = 18,
 } s2n_tls_extension_type;
 
+typedef enum {
+    S2N_TLS_MAX_FRAG_LEN_EXT_NONE,     /*!< don't use the extension    */
+    S2N_TLS_MAX_FRAG_LEN_512,          /*!< MaxFragmentLength 2^9      */
+    S2N_TLS_MAX_FRAG_LEN_1024,         /*!< MaxFragmentLength 2^10     */
+    S2N_TLS_MAX_FRAG_LEN_2048,         /*!< MaxFragmentLength 2^11     */
+    S2N_TLS_MAX_FRAG_LEN_4096,         /*!< MaxFragmentLength 2^12     */
+    S2N_TLS_MAX_FRAG_LEN_INVALID,      /*!< first invalid value        */
```

Comment:
After a more thorough pass at the PR I see why `S2N_TLS_MAX_FRAG_LEN_INVALID` is needed for internal implementation reasons. From a user-facing perspective, it is not needed. It would be nice to remove it, but if it is significantly painful I am okay with leaving it.